core/domain/android_validation_constants.py

```diff
@@ -43,3 +43,5 @@
 MAX_CHARS_IN_CHAPTER_TITLE = 36
 MAX_CHARS_IN_CHAPTER_DESCRIPTION = 152
 MAX_CHARS_IN_MISCONCEPTION_NAME = 100
+MAX_CHARS_IN_STORY_URL_FRAGMENT = 30
```

Comment:
Is this the max number of chars for an actual URL or just some part of the URL? Can this name be more specific to reflect the thing it's actually limiting?

Reply:
Hi @BenHenning -- I can help answer this one (a bit). The story URL fragment is the part of the URL that comes after "/story" -- e.g. in /learn/math/fractions/story/bakery/...., "bakery" is the "story URL fragment". (Similarly, "fractions" is the topic URL fragment, "learn" is the classroom URL fragment", and so on.)

I had some trouble with coming up with a name (besides "fragment") to represent a "word" in a URL. Do you have any suggestions?

Reply:
Thanks @seanlip! I've added these details in a comment above these constants for now.

Reply:
Ah I see. This isn't really an Android validation then, is it? This isn't actually a limitation that needs to be enforced on the Android side--the app will use whatever URLs it's told to.

FWIW I think 'fragment' is sufficiently clear.

Reply:
Hi @BenHenning I have added it in this file because this is going to be a required field that users will have to input while creating a new story / subtopic and so it will be a common field for android as well as web (I've added the same in constants.ts as well). Do you think I should have this defined only in constants.ts and not here?

Reply:
It doesn't need to be defined here, for the reason @BenHenning gave. This file should only contain stuff which affects UI display and general functionality on Android. (Could you please update its fileoverview to be clearer perhaps?)

Also, in general, never define the same constant in two different places. Otherwise you will get skew. If, at any time in the future, you feel you can't avoid this, feel free to chat with me and we can figure it out.

Reply:
(Oh -- just noticed you had a backend test for equality, which is a reasonable way to go about this if you have duplicate constants. Sorry, didn't see it before -- ignore the second paragraph of my last comment. Thanks!)

Reply:
Perfect, thanks!